### PR TITLE
Adds support for checksum endpoint.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
For use by Argo's checksum report job.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, run ***[integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** on stage as it tests preservation, and/or test in stage environment, in addition to specs.⚡

Unit, QA

### CODE REVIEW

File: spec/preservation/client/objects_spec.rb [hunk 1]
```diff
@@ -188,6 +188,38 @@
...
+          'sha256' => '6b66cc2df50427d03dca8608af20b3fd96d76b67ba41c148901aa1a60527032f',
+          'filesize' => '4403882'
+        }]
+      end
+
+      before do
+        allow(client).to receive(:get_json).with('objects/oo000oo0000/checksum', 'oo000oo0000').and_return(valid_json_response)
+      end
+
+      it 'returns the API response' do
```
Comment (ndushay) [inline spec/preservation/client/objects_spec.rb:207]: "returns valid json response??"

**Review by aaron-collier — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -74,6 +74,7 @@ Note that the preservation service is behind a firewall.
 
 - `client.objects.checksums(druids: druids)` - returns info as raw csv
 - `client.objects.checksums(druids: druids, format: 'json')` - returns info as json
+- `client.objects.checksum(druid: 'oo000oo0000')` - returns info as array of hashes
```
Comment (ndushay) [inline README.md:77]: ok, i'm late to the party here -- does the response indicate the resources that the hashes belong to?  Or the particular files?   It strikes me as too easy to be off.  But what do I know.

**Review by ndushay — COMMENTED** [review]
left a couple comments.

File: README.md (hunk 2 above)
Comment (justinlittman) [reply README.md:77]: In general, I think that the README gives a hint, but for full details a dev will need to consult Pres Cat's openapi.yml.

**Pull request merged**